### ISSUE TITLE
Feature request: add posibility to pass regular expressions to the 'import-order' rules

### ISSUE THREAD

**Issue by AndrewShapel** [benmosher/eslint-plugin-import#1015]
Hi everyone,

Now we have default groups, such as `["builtin", "external", "parent", "sibling", "index"]`, but would be better to add possibility to pass regular expressions to these groups definitions. For example, like this:

```
["buildin", "external", ".constants.js$", "index"]
```

It will improve code readability by adding customization of import groupings.
I just want to get approve from maintainers of this repository. Its not a problem to implement it and create PR by myself.

**Comment by ljharb** [comment]
The current purpose of the order rule is to group by *kind* - what you're asking for would be to group by *path*. I'm not sure if that's a particularly useful enhancement.

**Comment by AndrewShapel** [comment]
@ljharb Thank for your response.
So, if you think that this proposal is unnecessary, lets close this issue.

**Comment by AndrewShapel** [comment]
@ljharb @coreyfarrell I agree with this idea. In that case, we group imports by _kind_ and definition of these groups will be _paths_.

**Comment by atty-stripe** [comment]
Regex-defined import groups would be very helpful, especially in codebases that use path aliases (such as in Webpack.) 

I would love to require that all imports from a particular path alias are grouped together, as follows:

```
import moment from 'moment';

import a from 'foo-files/path/to/a.js';
import b from 'foo-files/path/to/b.js';

import x from 'bar-files/path/to/x.js';
import y from 'bar-files/path/to/y.js';
```

**Comment by gpbl** [comment]
I also would like to have more granular grouping since we use aliases via the [module-resolver](https://github.com/tleunen/babel-plugin-module-resolver#getting-started) babel plugin. 🙏

**Comment by munkyjunky** [comment]
A custom group would be very helpful - currently I'm working in a monorepo that has several packages included. When importing one package into another we want to import by `builtin`, `external`, `custom-packages`, `parent`, `sibling`, `index`

These custom packages are installed using `link-local` so they're currently seen as external modules. Being able to capture these as a custom group via regex would be a huge help!

**Comment by soryy708** [comment]
Related: https://github.com/benmosher/eslint-plugin-import/issues/674

**Comment by oliverdolgener** [comment]
I'd really like to see this merged. Such a good idea

**Comment by ZennKa** [comment]
+1 
would be really nice to have such feature

**Comment by bschlenk** [comment]
I'd like to be able to define a `css` group, anything ending with `/\.s?css$/` should be sorted last.

**Comment by ljharb** [comment]
Between #1570 and #1386 this seems like maybe it's already possible?
